lib/realtime/vehicles.ex

```diff
@@ -15,39 +15,54 @@ defmodule Realtime.Vehicles do
 
     # We show vehicles incoming from another route if they'll start the new route within 15 minutes
     incoming_trips = Schedule.active_trips(now, in_fifteen_minutes)
-    incoming_blocks_by_route = incoming_blocks_by_route(incoming_trips)
     # We show pulling out ghosts if they'll start within 10 minutes
     active_and_incoming_blocks_by_date = Schedule.active_blocks(now, in_ten_minutes)
 
-    group_by_route_with_blocks(
-      ungrouped_vehicles,
-      incoming_blocks_by_route,
-      active_and_incoming_blocks_by_date,
-      now
-    )
+    result =
+      group_by_route_with_blocks(
+        ungrouped_vehicles,
+        incoming_trips,
+        active_and_incoming_blocks_by_date,
+        now
+      )
+
+    # TODO: don't forget to take this bit out before you merge
+    result
+    |> Enum.map(fn {k, v} ->
+      {k, v |> Enum.map(& &1.route_id) |> Enum.filter(&(&1 != k))}
+    end)
+    |> Enum.reject(fn {_k, v} -> length(v) == 0 end)
+    |> IO.inspect()
+
+    result
   end
 
   @doc """
   Exposed for testing
   """
   @spec group_by_route_with_blocks(
           [Vehicle.t()],
-          Route.by_id([Block.id()]),
+          [Trip.t()],
           %{Date.t() => [Block.t()]},
           Util.Time.timestamp()
         ) ::
           Route.by_id([VehicleOrGhost.t()])
   def group_by_route_with_blocks(
         ungrouped_vehicles,
-        incoming_blocks_by_route,
+        incoming_trips,
         active_and_incoming_blocks_by_date,
         now
       ) do
     ghosts = Ghost.ghosts(active_and_incoming_blocks_by_date, ungrouped_vehicles, now)
     vehicles_and_ghosts = ghosts ++ ungrouped_vehicles
 
     incoming_from_another_route =
```

Comment:
The `incoming_from_another_route` also needs to be sorted with the laying over for which one will leave first.

Reply:
@skyqrose I'm not clear on how laying-over vehicles are involved here: they aren't displayed in the incoming box among  incoming vehicles, so their order wouldn't affect anything about how the incoming vehicles are displayed.

Reply:
Oops. `s/laying over/pulling out`. The incoming box shows two kinds of buses. Buses that are incoming from another route, and buses that are pulling out on that route. Both those categories should be sorted together.

(Sorry, I managed to have important factual errors in both my substantive comments 🤦 )